File: apps/android/app/src/main/java/com/clipbridge/app/Protocol.kt
package com.clipbridge.app

import org.json.JSONObject
import java.util.UUID

object Protocol {
    fun clipboardUpdate(fromDeviceId: String, content: String): String {
        return JSONObject()
            .put("version", 1)
            .put("type", "clipboard.update")
            .put("messageId", UUID.randomUUID().toString())
            .put("fromDeviceId", fromDeviceId)
            .put("toDeviceId", "windows")
            .put("contentType", "text")
            .put("content", content)
            .put("timestamp", System.currentTimeMillis())
            .toString()
    }

    fun hello(fromDeviceId: String): String {
        return JSONObject()
            .put("version", 1)
            .put("type", "device.hello")
            .put("messageId", UUID.randomUUID().toString())
            .put("fromDeviceId", fromDeviceId)
            .put("toDeviceId", "windows")
            .put("timestamp", System.currentTimeMillis())
            .toString()
    }
}

File: apps/windows/src/protocol.ts
export type ContentType = 'text';

export type ProtocolMessage = {
  version: 1;
  type:
    | 'pairing.request'
    | 'pairing.confirm'
    | 'pairing.reject'
    | 'device.hello'
    | 'device.ping'
    | 'clipboard.update'
    | 'clipboard.ack'
    | 'error';
  messageId: string;
  fromDeviceId: string;
  toDeviceId?: string;
  contentType?: ContentType;
  content?: string;
  timestamp: number;
};

export type PairingInfo = {
  app: 'clipbridge';
  version: 1;
  host: string;
  port: number;
  pairingCode: string;
  deviceName: string;
  expiresAt: number;
};

export type HistoryItem = {
  id: string;
  direction: 'sent' | 'received';
  sourceDevice: string;
  contentType: ContentType;
  content: string;
  timestamp: number;
};
